File: python/enforced_client.py
import json

from flask import Flask
from oso_cloud import Oso

app = Flask(__name__)

api_key = "<please provide your api key here>"
oso = Oso(url="https://cloud.osohq.com", api_key=api_key)


@app.route("/organization/<org_id>")
def get(org_id):
    actor = {"type": "User", "id": "anonymous"}
    resource = {"type": "Organization", "id": org_id}

    if not oso.authorize(actor, "view", resource):
        return "Not Found", 404

    return json.dumps({"id": org_id}), 200


if __name__ == "__main__":
    app.run(port=8000)

File: node/enforced_client.js
const express = require("express");
const { Oso } = require("oso-cloud");

async function start() {
  const app = express();

  const apiKey =
    "<please provide your api key here>";
  const oso = new Oso("https://cloud.osohq.com", apiKey);

  app.get("/organization/:org_id", async (req, res) => {
    const orgId = req.params.org_id;

    const actor = { type: "User", id: "anonymous" };
    const resource = { type: "Organization", id: orgId };
    if ((await oso.authorize(actor, "view", resource)) === false) {
      // Handle authorization failure
      res.status(404).send("Not Found");
      return;
    }

    res.status(200).send(`{"id": ${orgId}}`);
  });
  app.listen(8000);
}

start();
